File: server/src/main/java/br/app/pregsrateio/usuario/api/dto/CadastroUsuarioRequest.java
package br.app.pregsrateio.usuario.api.dto;

import br.app.pregsrateio.usuario.data.Usuario;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CadastroUsuarioRequest {

    private String nome;
    private String sobrenome;
    private String email;
    private String numeroCelular;
    private String fotoUrl;

    public Usuario toUsuario(String authSub) {
        return Usuario.builder()
                .authSub(authSub)
                .nome(nome)
                .sobrenome(sobrenome)
                .email(email)
                .numeroCelular(numeroCelular)
                .fotoUrl(fotoUrl)
                .build();
    }
}

File: server/src/main/kotlin/br/dev/brunorsch/pregsrateio/usuario/api/dto/CadastroUsuarioRequest.kt
package br.dev.brunorsch.pregsrateio.usuario.api.dto

import br.dev.brunorsch.pregsrateio.usuario.model.Usuario
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class CadastroUsuarioRequest(
    val nome: String,
    val sobrenome: String,
    val email: String,
    val whatsApp: String,
    val fotoUrl: String? = null,
) {
    fun toUsuario(authSub: String) = Usuario(
        authSub = authSub,
        nome = nome,
        sobrenome = sobrenome,
        email = email,
        whatsApp = whatsApp,
    )
}